src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptions.cs

```diff
@@ -1,23 +1,59 @@
 ﻿namespace Microsoft.Extensions.DependencyInjection
 {
     using System.Collections.Generic;
+    using System.Linq;
     using Microsoft.ApplicationInsights.Extensibility;
     using Microsoft.Extensions.Options;
 
     /// <summary>
-    /// The <see cref="IOptions{TelemetryConfiguration}"/> implementation that uses <see cref="TelemetryConfiguration.Active"/> as initial value.
+    /// The <see cref="IOptions{TelemetryConfiguration}"/> implementation that create new <see cref="TelemetryConfiguration"/> every time when called".
     /// </summary>
     internal class TelemetryConfigurationOptions : IOptions<TelemetryConfiguration>
     {
         public TelemetryConfigurationOptions(IEnumerable<IConfigureOptions<TelemetryConfiguration>> configureOptions)
         {
-            foreach (var c in configureOptions)
+            // workaround for Microsoft/ApplicationInsights-dotnet#613
+            this.Value = TelemetryConfiguration.CreateDefault();
+
+            var configureOptionsArray = configureOptions.ToArray();
+            foreach (var c in configureOptionsArray)
             {
                 c.Configure(this.Value);
             }
+
+            // as we expect some customers to use TelemetryConfiguration.Active together with dependency injection
+            // we make sure it has been set up, it must be done only once even if there are multiple Web Hosts in the process
+            if (!IsActiveConfigured(this.Value.InstrumentationKey))
+            { 
+                foreach (var c in configureOptionsArray)
+                {
+                    c.Configure(TelemetryConfiguration.Active);
+                }
+            }
         }
 
         /// <inheritdoc />
-        public TelemetryConfiguration Value => TelemetryConfiguration.Active;
+        public TelemetryConfiguration Value { get; }
+
+        /// <summary>
+        /// Determines if TelemetryConfiguration.Active needs to be configured.
+        /// </summary>
+        /// <param name="instrumentationKey">Instrumentation key.</param>
+        /// <returns>True is TelemertryConfiguration.Active was previously configured.</returns>
+        private static bool IsActiveConfigured(string instrumentationKey)
+        {
+            var active = TelemetryConfiguration.Active;
+            if (string.IsNullOrEmpty(active.InstrumentationKey) && !string.IsNullOrEmpty(instrumentationKey))
+            {
+                return false;
+            }
+
+            if (active.TelemetryInitializers.Count <= 1)
```

Comment:
This looks like a new enforced condition that we should have at least 2 telemetry initializers for Active to be configured. Should we add a test to actually require it (default TI > 2) and don't miss if some TIs are removed later?

Reply:
the other TI is coming from base SDK, so enforcing it with tests in aspnetcore SDK would not really help.

However if it ever changes, tests in aspnetcore SDK would fail for some other issues and we'd probably have to spend a lot of time figuring out why. I'd add a test with a comment to address this